presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/optimization/JdbcComputePushdown.java

```diff
@@ -100,18 +110,23 @@ public PlanNode visitFilter(FilterNode node, Void context)
             TableHandle oldTableHandle = oldTableScanNode.getTable();
             JdbcTableHandle oldConnectorTable = (JdbcTableHandle) oldTableHandle.getConnectorHandle();
 
+            RowExpression toTranslate = expressionOptimizer.optimize(node.getPredicate(), OPTIMIZED, session);
+            toTranslate = logicalRowExpressions.convertToConjunctiveNormalForm(toTranslate);
```

Comment:
Call this inside the `rowExpressionToSqlTranslator `; it's part of the filter logic

Reply:
Where? There is no clear entry-point in RowExpressionTranslator as it simply just visits the corresponding node it is called on.

Unless you mean that I should do this during visitSpecialForm?

Reply:
Leaving as is after in-person sync.